v0.5/p1/spec/EBNF.md

```diff
@@ -0,0 +1,38 @@
+# EBNF 
+
+```chatinput
+<fastn> ::= <comment>* <module_comment>? <item>*
+<item> ::=  <comment> | <section>
+
+
+<module_comment> ::= (<single_module_comment> <new_line>)* <single_module_comment> (<new_line> | <eof>)
+<single_module_comment> ::= <doc_comment_marker> <opt_whitespace> <sentence>? <opt_whitespace> 
+
+<comment> ::= (<single_comment> <new_line>)* <single_comment> (<new_line> | <eof>)
+<single_comment> ::= <comment_marker> <opt_whitespace> <sentence>? <opt_whitespace>
+
+
+<section> ::= <start_marker> <opt_whitespace> <section_declaration>? <new_line>?
+<section_declaration> ::= (<section_kind> <whitespace>)? <section_name> <colon> <opt_whitespace> <section_value>? <opt_whitespace>
+<section_kind> ::= <word>
+<section_name> ::= <word>
+<section_value> ::= <word>
+
+
+
+<start_marker> ::= "--"
```

Comment:
Let's rename this to `section_start_marker`. Should we include the space here itself as that is required part of our syntax we just discussed?

Reply:
Also I think the grammar would be easier to read if we do not define names for things that are ever used once.